psydac/linalg/tests/test_block.py

```diff
@@ -170,6 +171,37 @@ def test_block_matrix( n1, n2, p1, p2, P1, P2  ):
 
     assert np.allclose( y, yref, rtol=1e-12, atol=1e-12 )
 
+#===============================================================================
+@pytest.mark.parametrize( 'n1', [8,16] )
+@pytest.mark.parametrize( 'n2', [8,12] )
+@pytest.mark.parametrize( 'p1', [1,2,3] )
+@pytest.mark.parametrize( 'p2', [1,2,3] )
+@pytest.mark.parametrize( 'P1', [True, False] )
+@pytest.mark.parametrize( 'P2', [True, False] )
+
+def test_block_2d_array_to_stencil( n1, n2, p1, p2, P1, P2 ):
+
+    # Create vector spaces, and stencil vectors
+    V1 = StencilVectorSpace( [n1,n2], [p1,p2], [P1,P2] )
+    V2 = StencilVectorSpace( [n1,n2], [p1,p2], [P1,P2] )
+
+    W = BlockVectorSpace(V1, V2)
+
+    x = BlockVector(W)
+
+    # Fill in vector with random values, then update ghost regions
+    for i1 in range(n1):
+        for i2 in range(n2):
+            x[0][i1,i2] = 2.0*random() + 1.0
+            x[1][i1,i2] = 5.0*random() - 1.0
+    x.update_ghost_regions()
+
+    xa = x.toarray()
+    v  = array_to_stencil(xa, W)
+
+    assert np.allclose( xa , v.toarray() )
```

Comment:
I think that we also need to test the case where one of the two spaces `V1, V2` is a `BlockVectorSpace` itself. This is relevant to our multi-patch vector examples if I am not mistaken.

Reply:
done

Reply:
Thanks